tests/rendered_output.rs

```diff
@@ -467,6 +467,76 @@ fn by_default_mdbook_use_index_preprocessor_to_convert_readme_to_index() {
     assert_doesnt_contain_strings(&second_index, &unexpected_strings);
 }
 
+#[test]
+fn first_chapter_is_copied_as_index_even_if_not_first_elem() {
+    let temp = DummyBook::new().build().unwrap();
+    let mut cfg = Config::default();
+    cfg.set("book.src", "index_html_test")
+        .expect("Couldn't set config.book.src to \"index_html_test\"");
+    let md = MDBook::load_with_config(temp.path(), cfg).unwrap();
+    md.build().unwrap();
+
+    let chapter_path = temp.path().join("book").join("chapter_1.html");
+    let index_path = temp.path().join("book").join("index.html");
+    // In theory, just reading the entire files into memory and comparing is sufficient for *testing*,
```

Comment:
I think this could use `pretty_assertions::assert_eq` to get a highlighted context diff.  I think it would be good to change this to just read the files into memory and use `pretty_assertions`. Would that work?

Reply:
I agree that performance is not really a concern as far as failure is concerned, but I was kind of worried for a moment that it was frozen.

The real concern I had was that the entire file gets printed on a single line, which is not all too handy. But this is no longer the case with `pretty_assertions` v1.2, *and* it's fast, so indeed, no reason not to.

![](https://user-images.githubusercontent.com/15271137/176370077-1540cc9d-2523-4ab8-91f7-de4da9f82f51.png)